Add Nebius backend

jvstme opened the pull request:
Closes #2390

jvstme committed:
Add Nebius backend

r4victor commented on src/dstack/_internal/core/backends/nebius/compute.py:
```diff
@@ -0,0 +1,266 @@
+        if instance is not None:
+            if instance.status.state != instance.status.InstanceState.DELETING:
+                resources.delete_instance(self._sdk, instance_id)
+                raise WorkRemains(
```
I'd raise ComputeSpecific-errors and handle it accordingly instead of introducing generic errors. At least, it keeps the error hierarchy - Compute returns ComputeError.

jvstme committed:
Add docs and tests

jvstme committed:
`WorkRemains` -> `NotYetTerminated`

jvstme committed:
Handle quota errors with incorrect code

IP address quota violations result in the
INVALID_ARGUMENT error code instead of
RESOURCE_EXHAUSTED, but mention the violation in
the error message.

jvstme committed:
Replace sync SDK API with a custom wrapper

The sync API suffers from deadlocks, as stated in
the SDK readme. Deadlocks often happen during
concurrent actions, such as creating or deleting
fleets with many instances, and lead to a
dysfunctional dstack-server. This commit solves
the problem by switching to the async API and
running the async methods in a separate event loop
dedicated to Nebius.
